models/attachment.go

```diff
@@ -1,11 +1,149 @@
 package models
 
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
+
 // Attachment contains the fields and methods for
 // an email attachment
 type Attachment struct {
-	Id         int64  `json:"-"`
-	TemplateId int64  `json:"-"`
-	Content    string `json:"content"`
-	Type       string `json:"type"`
-	Name       string `json:"name"`
+	Id          int64  `json:"-"`
+	TemplateId  int64  `json:"-"`
+	Content     string `json:"content"`
+	Type        string `json:"type"`
+	Name        string `json:"name"`
+	vanillaFile bool   // Vanilla file has no template variables
+}
+
+// ValidateAttachment ensures that the provided attachment uses the supported template variables correctly.
+func (a Attachment) ValidateAttachment() error {
+
+	ptx := PhishingTemplateContext{
```

Comment:
For consistency, could we make sure we're using the same values as the existing [ValidateTemplate](https://github.com/gophish/gophish/blob/b01bd6cbc0bbf3109511d280b545708b3e2089d0/models/template_context.go#L101) function? I'm sure you can tell by now, but I quite like consistency where possible 😄 

I wish there were a way for us to reuse `ValidateTemplate` itself, but I think that might be too tricky.

Reply:
Good call, consistency is key! I did notice the values in [ValidateTemplate](https://github.com/gophish/gophish/blob/b01bd6cbc0bbf3109511d280b545708b3e2089d0/models/template_context.go#L101) are missing `Position` (in BaseRecipient), `URL`, `TrackingURL`, `Tracker`, and `From` parameters. I assume these are automatically handled - but haven't checked thoroughly.

Either way, I'll update to match - and maybe in the future we can roll the attachment validation into `ValidateTemplate`.